medicines/doc-index-updater/src/state_manager/redis.rs

```diff
@@ -72,13 +72,16 @@ pub async fn get_from_redis(client: Client, id: Uuid) -> RedisResult<JobStatus>
 pub async fn set_in_redis(client: Client, id: Uuid, status: JobStatus) -> RedisResult<JobStatus> {
     let mut con = client.get_async_connection().await?;
 
-    redis::cmd("SET")
-        .arg(id.to_string())
-        .arg(status.clone())
-        .query_async(&mut con)
-        .await?;
+    let lua_script = match status {
+        JobStatus::Accepted => include_str!("set_if_not_exists.lua"),
+        _ => include_str!("set_if_not_completed.lua"),
```

Comment:
This script will also not set if errored or in any other state so maybe `set_if_not_exists_or_accepted`?